snuba/datasets/querylog.py

```diff
@@ -0,0 +1,89 @@
+from datetime import timedelta
+from snuba import settings
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnSet,
+    DateTime,
+    Enum,
+    Float,
+    LowCardinality,
+    Mapping,
+    Nested,
+    String,
+    UInt,
+)
+
+from snuba.datasets.dataset import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.schemas.tables import MergeTreeSchema
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.datasets.querylog_processor import QuerylogProcessor
+from snuba.query.extensions import QueryExtension
+from snuba.query.project_extension import ProjectExtension, ProjectExtensionProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+
+
+class QuerylogDataset(TimeSeriesDataset):
+    def __init__(self) -> None:
+        status_type = Enum([("success", 0), ("error", 1), ("rate-limited", 2)])
+
+        columns = ColumnSet(
+            [
+                ("request", String()),
+                ("referrer", LowCardinality(String())),
+                ("dataset", LowCardinality(String())),
+                ("projects", Array(UInt(64))),
+                ("timestamp", DateTime()),
+                ("duration_ms", UInt(32)),
+                ("status", status_type),
+                (
+                    "clickhouse_queries",
+                    Nested(
+                        [
+                            ("sql", String()),
+                            ("status", status_type),
+                            ("final", UInt(8)),
+                            ("cache_hit", UInt(8)),
+                            ("sample", Float(32)),
+                            ("max_threads", UInt(8)),
```

Comment:
I can see us adding columns (and then dropping them) relatively often to track specific issues, I don't think we can afford to change the schema each time, so (to be discussed with the team) but I think a concept of tags to add/remove new fields quickly would be useful.

Some example I can think we will want to add:
- serialize the AST for debug
- list the query processors we applied 
- write a more structured representation of the query to be able to do some breakdowns.


Also we likely need trace id and span id to correlate with APM data.

Reply:
Yeah, that makes sense especially for ad hoc dataset specific stuff that might not be relevant for all logged queries. I'd rather add that when we actually need it though, I don't think adding tags later when we confirm that we need and want to go down that route will be too difficult.